src/signature.c

```diff
@@ -82,20 +100,38 @@ bool initialize_signature(void)
 
 	ret = validate_certificate();
 	if (ret) {
-		printf("Failed to verify certificate: %s\n", X509_verify_cert_error_string(ret));
 		if (ret == X509_V_ERR_CERT_NOT_YET_VALID) {
-			/* If we can retrieve an approx. good system time, report out to user */
-			if (stat("/usr/lib/os-release", &statt) != -1) {
-				mod_sec = statt.st_mtim.tv_sec;
-				char timebuf[30];
-				alttime = localtime(&mod_sec);
-				strftime(timebuf, sizeof(timebuf), "%F", alttime);
-				printf("System clock should be at least %s\n", timebuf);
+			time_t currtime;
```

Comment:
Then we should also check if the system time is ahead of the file's date as well. This check is a bit annoying, and at worst we'd be setting a sane system time to a still sane system time. If the cert is bad, is there any case where it should not fail anyway?